hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java

```diff
@@ -60,8 +66,23 @@ public class HiveCreateReplaceTableTest extends HiveMetastoreTest {
 
   private String tableLocation;
 
+  @RegisterExtension
+  private static final HiveMetastoreExtension HIVE_METASTORE_EXTENSION =
+      new HiveMetastoreExtension(DB_NAME, Collections.emptyMap());
+
+  private HiveCatalog catalog;
+
   @BeforeEach
   public void createTableLocation() throws IOException {
+    catalog =
```

Comment:
I think we should initialize the catalog only once. Not for each tests as nothing changes between tests related to catalog config.

Reply:
Also can we move this to `HiveMetastoreExtension` and get the catalog from there? It can help in duplicating this code in each test class which uses this extension.

Reply:
> Also can we move this to `HiveMetastoreExtension` and get the catalog from there? It can help in duplicating this code in each test class which uses this extension.

I don't think the catalog initialization should be moved into `HiveMetastoreExtension`. The sole purpose of `HiveMetastoreExtension` is to provide the metastore part

Reply:
@nastra: Thoughts?

Reply:
I've replied already above

Reply:
Sorry, I wanted your thoughts on this comment
https://github.com/apache/iceberg/pull/9282#discussion_r1423803359